### PR TITLE
Fixed bundling ReactDOM

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,31 +1,29 @@
 {
   "name": "react-final-form-html5-validation",
   "version": "1.0.0",
-  "description":
-    "A swap-in replacement for 🏁 React Final Form's <Field> component to provide HTML5 Validation",
+  "description": "A swap-in replacement for 🏁 React Final Form's <Field> component to provide HTML5 Validation",
```
Comment (Andarist) [inline package.json]: I think my npm formats package.json when doing a fresh install for some reason, no idea how to stop it (havent searched for it though 😉 ). Hope it's ok

File: package.json [hunk 2]
```diff
@@ -71,13 +69,19 @@
     "final-form": ">=4.0.0",
     "prop-types": "^15.6.0",
     "react": "^15.3.0 || ^16.0.0-0",
+    "react-dom": "^15.3.0 || ^16.0.0-0",
```
Comment (Andarist) [inline package.json]: this was added! this peer dep was already required, only not in `package.json`

File: rollup.config.js [hunk 3]
```diff
@@ -49,18 +53,19 @@ export default {
...
     },
     output
   ),
-  external: ['react', 'prop-types', 'final-form', 'react-final-form'],
+  external: umd ? peers : deps.concat(peers),
   plugins: [
     resolve({ jsnext: true, main: true }),
     flow(),
-    commonjs({ include: 'node_modules/**' }),
+    umd && commonjs({ include: 'node_modules/**' }),
```
Comment (Andarist) [inline rollup.config.js]: just extra guard, for non-umd build u dont want to use this plugin at all

File: package.json [hunk 4]
```diff
@@ -69,7 +69,6 @@
     "final-form": ">=4.0.0",
     "prop-types": "^15.6.0",
     "react": "^15.3.0 || ^16.0.0-0",
-    "react-dom": "^15.3.0 || ^16.0.0-0",
```
Comment (Andarist) [inline package.json]: this should be kept

Comment (Andarist) [reply package.json]: CI is failing now because this got removed

**Comment by erikras** [comment]
Any proof that this reduces bundle size? My initial test was pretty equivalent to `master`.

**Comment by Andarist** [comment]
I've made tests after running `./node_modules/.bin/nps build.es` and then inspecting the size with `wc -c dist/react-final-form-html5-validation.es.js`

master 670401
my PR 6402

Rollup build times also went down quite a bit (1.8s vs 422ms)

**Comment by Andarist** [comment]
The problem was using [`findDOMNode`](https://github.com/final-form/react-final-form-html5-validation/blob/3f213212066d28daa6b749dbdea609083150506a/src/Html5ValidationField.js#L42) but not specifying `react-dom` as external

**Comment by erikras** [comment]
Um, confirmed. From 30 kb down to 1.45kb. 🤦‍♂️💥

**Comment by erikras** [comment]
Published in [`v1.0.1`](https://github.com/final-form/react-final-form-html5-validation/releases/tag/v1.0.1).

`dist/react-final-form-html5-validation.es.js` went from `730K` to `5.4K`. 😱

**Pull request merged**